schemas/expconf/v0/profiling.json

```diff
@@ -31,10 +31,12 @@
         }
     },
     "conditional": {
-        "$comment": "when enabled=true, assert begin <= end",
+        "$comment": "when enabled=true and both begin and end are present, assert begin <= end",
```

Comment:
Ok, so based on the behavior you are describing, I actually don't think the conditional needs to be present at all.

Thinking about it, the comment says `when enabled=true, assert begin <= end`... but actually that's not accurate anymore.  Actually the `compareProperties` subschema could be outside of this conditional and work just fine.  `compareProperties` already ignores situations where one or both properties are not defined, so it doesn't need to be inside a `conditional` at all.

Really, what the old schema did was make both properties required only when `enabled == true`.  (since you asked, the `properties.enabled = {const: true}` thing only matches when the `enabled` property matches the constant value `true`.  The `when` clause also _required_ `enabled`, otherwise an empty object would still match, because that's how json-schema matching works).

But, if the correct behavior is for `begin_on_batch` and `end_after_batch` to *never* be required, you can basically replace the whole `condition` clause with just the `compareProperties` clause, and adjust the defaults accordingly.

Don't forget to edit the existing tests and add more tests in `schemas/tests_cases/v0/misc.yaml`; the profiling test cases are all in a clump at the bottom.

Reply:
(thinking more about it, I think the `conditional` clause _was_ initially necessary before `end_on_batch` was re-written as `end_after_batch`, and the old `compareProperties` was of `type: "a<b"` which failed on pre-expconf schemas that had zero defaults)